### PR TITLE
feat(console): application details settings

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): application details settings
### Todo
* LOG-1686: "Add another" in Application Detail Page
* LOG-1678: Save Changes

## Linear Issue Reference

LOG-1674

## Testing

image
image

### CODE REVIEW

File: packages/console/src/pages/ApplicationDetails/index.tsx [hunk 1]
```diff
@@ -1,46 +1,149 @@
...
 import { RequestError } from '@/swr';
 import { applicationTypeI18nKey } from '@/types/applications';
 
 import * as styles from './index.module.scss';
 
+type OidcConfig = {
+  authorization_endpoint: string;
+  userinfo_endpoint: string;
+  token_endpoint: string;
+};
```
Comment (demonzoo) [inline packages/console/src/pages/ApplicationDetails/index.tsx:25]: Would it be better to define this type in `schemas` package?

File: packages/console/src/pages/ApplicationDetails/index.tsx [hunk 2]
```diff
@@ -1,46 +1,149 @@
...
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
-  const isLoading = !data && !error;
+
+  const isNormalSettings = !location.pathname.includes('advanced-settings');
+
+  const buildTabLink = (tabLinkName: string) => {
+    const index = location.pathname.lastIndexOf('/');
+
+    if (index < 0) {
+      return location.pathname;
```
Comment (demonzoo) [inline packages/console/src/pages/ApplicationDetails/index.tsx:50]: This looks weird. Should it be as follows?
```typescript
return `${location.pathname}/${tabLinkName}`;
```

File: packages/console/src/pages/ApplicationDetails/index.tsx [hunk 3]
```diff
@@ -1,46 +1,149 @@
...
+    }
+
+    return `${location.pathname.slice(0, Math.max(0, index))}/${tabLinkName}`;
```
Comment (demonzoo) [inline packages/console/src/pages/ApplicationDetails/index.tsx:53]: If you want to remove the last `/`, you can use
```typescript
.slice(0, -1)
```

File: packages/console/src/pages/ApplicationDetails/index.tsx [hunk 4]
```diff
@@ -1,46 +1,149 @@
...
+                        <TextInput {...register('description')} />
+                      </FormField>
+                      <FormField title="admin_console.application_details.authorization_endpoint">
+                        <CopyToClipboard
+                          className={styles.copy}
+                          value={oidcConfig.authorization_endpoint}
+                        />
+                      </FormField>
+                    </>
+                  ) : (
```
Comment (demonzoo) [inline packages/console/src/pages/ApplicationDetails/index.tsx:119]: Should avoid using long conditional operator expression. Try break it into 2 parts
```typescript
{isNormalSettings && <>...</>}
{!isNormalSettings && <>...</>}
```

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 1 above)
Comment (IceHe) [reply packages/console/src/pages/ApplicationDetails/index.tsx:25]: extract to the other place other than `schemas` package?

Comment (IceHe) [reply packages/console/src/pages/ApplicationDetails/index.tsx:25]: better re-use `OidcConfigSnakeCaseResponse` in `js/js` package?

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 2 above)
Comment (IceHe) [reply packages/console/src/pages/ApplicationDetails/index.tsx:50]: better use `new URL()` to add a slash or remove duplicate ones

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx:53]: I just want to replace the last path segment with the correct tab name.
We have a path: `https://www.logto.dev/console/applications/some-id/settings`.
What I want to do is remove the last `/settings` and change it to `/advanced-settings`.
Finally, We get `https://www.logto.dev/console/applications/some-id/advanced-settings`.

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx:50]: No. See the reply below.
https://github.com/logto-io/logto/pull/381#discussion_r825757803

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx:25]: This is not a schema.
It returns from the oidc provider and currently, we can not share the codes between the `core` and the `js` package.
@IceHe

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx]: updated

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx]: updated

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx]: updated

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 1 above)
Comment (demonzoo) [reply packages/console/src/pages/ApplicationDetails/index.tsx:25]: OK, I see. We can consider exporting this type from js sdk, as we are going to integrate React sdk in #373. So technically we can export this type all the way from core js sdk to React sdk.
I can add a task to do that tomorrow.

Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx:25]: Added: LOG-1876

File: packages/console/src/App.tsx [hunk 5]
```diff
@@ -44,7 +44,9 @@ const Main = () => {
             <Routes>
               <Route path="applications">
                 <Route index element={<Applications />} />
-                <Route path=":id" element={<ApplicationDetails />} />
+                <Route path=":id" element={<Navigate to="settings" />} />
+                <Route path=":id/settings" element={<ApplicationDetails />} />
```
Comment (wangsijie) [inline packages/console/src/App.tsx:48]: can be `:id`'s sub route?

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/ApplicationDetails/index.tsx:25]: Is just a form type, I think is OK to define a type here.

**Review by wangsijie — APPROVED** [review]
Overall LGTM

**Review by demonzoo — APPROVED** [review]

File: packages/console/src/App.tsx (hunk 5 above)
Comment (xiaoyijun) [reply packages/console/src/App.tsx]: updated

**Pull request merged**